src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -220,7 +222,10 @@ private static IVecInt toVecInt(IntStream stream) {
 					}
 				}
 
-				// assume satisfied
+				// TODO: better error message, have a way to figure out precisely which mods are causing this
+				if (!problem.isSatisfiable(assumptions)) {
+					throw new ModResolutionException("Could not resolve mod collection due to an unknown error");
```

Comment:
How about printing the problem?

Reply:
Oh yeah, that might be helpful. I'm not sure what form it'd appear in, so we'll have to see.

Reply:
I dont think this is much better than the real error, fixing the issue will be best

Reply:
Hmm, I just want you to print more info on the unsatisfied problem either in log or in exception message..

Reply:
I have not been able to confirm that problems have a satisfactory `toString` implementation.

Reply:
Sure, it you don't print the problem guess we can attach a debugger at this throw breakpoint and just look up the problem from the debugger.